opm/simulators/linalg/MILU.cpp

```diff
@@ -33,19 +33,19 @@
 #include <array>
 
 template <typename T>
-T Opm::detail::identityFunctor(const T& t){return t;};
+std::function<T(const T&)> Opm::detail::identityFunctor(const T& t){return t;};
 
 template <typename T>
-T Opm::detail::oneFunctor(const T&){return 1.0;};
+std::function<T(const T&)> Opm::detail::oneFunctor(const T&){return 1.0;};
 
 template <typename T>
-T Opm::detail::signFunctor(const T& t){if (t< 0){return -1;} else{return 1;}};
+std::function<T(const T&)>  Opm::detail::signFunctor(const T& t){if (t< 0){return -1;} else{return 1;}};
 
 template <typename T>
-T Opm::detail::isPositiveFunctor(const T& t){if (t<0){return 0;} else{return 1;}};
+std::function<T(const T&)>  Opm::detail::isPositiveFunctor(const T& t){if (t<0){return 0;} else{return 1;}};
 
 template <typename T>
-T Opm::detail::absFunctor(const T& t){return std::abs(t);};
+std::function<T(const T&)>  Opm::detail::absFunctor(const T& t){return std::abs(t);};
```

Comment:
Checked! 
Now templates are a bit mixed-type, in the sense that `oneFunctor`, `signFunctor`, and `isPositiveFunctor` return double and take `const T&`

I'm compiling a second time because there was an ambiguous thing between my template declarations in the header file and template definitions in the source file.